lib/matplotlib/tests/test_lines.py

```diff
@@ -129,6 +129,18 @@ def test_valid_drawstyles():
     with pytest.raises(ValueError):
         line.set_drawstyle('foobar')
 
+        
+def test_set_drawstyle():
+    x = np.linspace(0, 2*np.pi, 10)
+    y = np.sin(x)
+
+    fig, ax = plt.subplots()
+    line, = ax.plot(x, y)
+    line.set_drawstyle("steps-pre")
+    assert len(line.get_path().vertices) == 2*len(x)-1
+    
+    line.set_drawstyle("default")
+    assert len(line.get_path().vertices) == len(x)
 
```

Comment:
PEP8 will complain you need an extra line here (and maybe some other things above - I'm terrible at PEP8.  But once this passes it looks good to me.  Thanks!